### PR TITLE
fix: Fix Asset generation which otherwise fails due to unknown resources

### PR DESCRIPTION
**Pull request by jskeet**
Before this commit, generation fails if any resource name field is
loaded without the resource definition being present. That's only
actually a problem if the field is used for generation - otherwise,
it's fine.

This commit fixes Google.Cloud.Asset.V1 to avoid an
otherwise-unnecessary import being required.

Once this has been submitted, we can create a release which will fix the Kokoro builds for googleapis/googleapis, and we can remove the workaround in google-cloud-dotnet.

### CODE REVIEW

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Comment by jskeet** [comment]
Thanks Amanda - I'll wait for @viacheslav-rostovtsev to review as well, given that this is a sensitive area.

**Comment by jskeet** [comment]
Merging now to unblock internal CI.

**Pull request merged**
